cosmic-settings/src/pages/power/mod.rs

```diff
@@ -47,6 +53,45 @@ impl Page {
     }
 }
 
+fn battery_info() -> Section<crate::pages::Message> {
+    let descritpions = Slab::new();
+
+    Section::default()
+        .title(fl!("battery"))
+        .descriptions(descritpions)
+        .view::<Page>(move |_binder, _page, section| {
+            let runtime = tokio::runtime::Runtime::new().unwrap();
+            let battery = runtime.block_on(Battery::update_battery());
```

Comment:
This should be returned by the update method in a `Command`.

```rs
return cosmic::command::future(async move {
    let battery = Battery::update_battery().await;
    Message::UpdateBattery(battery)
})
.map(crate::pages::Power)
.map(crate::app::PageMessage);

Reply:
Oh, perfect! That's was my though but I was struggling with it a bit so just followed power profiles implementation. Gonna use `fn on_enter` and this command :)